slf4j/src/main/scala/org/typelevel/log4cats/slf4j/internal/Slf4jLoggerInternal.scala

```diff
@@ -77,14 +73,19 @@ private[slf4j] object Slf4jLoggerInternal {
   }
 
   @nowarn("msg=used")
-  final class Slf4jLogger[F[_]](val logger: JLogger, sync: Sync.Type = Sync.Type.Delay)(implicit
-      F: Sync[F]
-  ) extends SelfAwareStructuredLogger[F] {
+  final class Slf4jLogger[F[_]](
+      val logger: JLogger,
+      sync: Sync.Type = Sync.Type.Delay,
+      defaultCtx: F[Map[String, String]]
```

Comment:
By getting the default context into the logger, as an effect, we are now able to avoid the asking and concatenation of the context map when the log is disabled.  But this is a bit less conducive to the middleware approach: we don't want `addContext` to return a `ModifiedContextStructuredLogger` anymore.